webapp/app/js/controllers/sourceMeta.js

```diff
@@ -131,6 +131,39 @@ KylinApp
       });
     };
 
+    $scope.calCardinality = function (tableName) {
+      SweetAlert.swal({
+        title: "",
+        text: "Are you sure to recalculate column cardinality?",
+        showCancelButton: true,
+        confirmButtonColor: '#DD6B55',
+        confirmButtonText: "Yes",
+        cancelButtonText: "No",
+        closeOnConfirm: true
+      }, function (isConfirm) {
+        if (isConfirm) {
+          if (!$scope.projectModel.selectedProject) {
+            SweetAlert.swal('', 'Please choose your project first!.', 'info');
```

Comment:
The message should be consistent with other parts in below: "Please select a project." And, please avoid using "!" in the sentence as its mood is too strong.

Reply:
OK, 'Please choose your project first!.' is also used in the other two places. I will deal with them together